File: Leetcode/JS/Leetcode382-链表随机节点.js
/**
 * @param {ListNode} head
 */
var Solution = function(head) {
    this.node = []
    while(head){
        this.node.push(head.val);
        head = head.next;
    }
};

/**
 * @return {number}
 */
Solution.prototype.getRandom = function() {
    return this.node[Math.floor(Math.random() * this.node.length)];
};

File: Leetcode/sbgo/Leetcode382-链表随机节点.go
package sbgo

import "math/rand"

type Solution struct {
	node []int
}

func Constructor(head *ListNode) Solution {
	t := []int{}
	for head != nil {
		t = append(t, head.Val)
		head = head.Next
	}
	return Solution{t}
}

func (this *Solution) GetRandom() int {
	return this.node[rand.Intn(len(this.node))]
}
